### PR TITLE
[ch10754] Location filtering and search

### PR DESCRIPTION
**Pull request by AttilaTheHen**
##### Feature list
* _Describe the list of features from Django_
  * Add pagination and filter for location title
* _Describe the list of features from Polymer_
  * Have location dropdowns populate with the paginated location results
  * Enable searching through all location results and populate the dropdown
  * Change styling to have dropdown height be fixed

##### Progress checker
- [x] Django

- [x] Polymer

### CODE REVIEW

**Comment by n1207n** [comment]
This PR completes [ch10754]

File: polymer/src/elements/cluster-reporting/indicator-locations-widget.html [hunk 1]
```diff
@@ -357,6 +371,45 @@ <h3>Locations ([[value.length]])</h3>
           },
         },
 
+        _onValueChanged: function(event) {
+          var loc_type = this.get('searchLocationType');
+
+          var self = this;
+          if(event.detail.value === '') {
+            this.$$('#locations' + loc_type).thunk()()
```
Comment (n1207n) [inline polymer/src/elements/cluster-reporting/indicator-locations-widget.html]: Looks like we are triggering thunk right away if search text is empty. Should we wrap this around with debounce?

File: polymer/src/elements/cluster-reporting/indicator-locations-widget.html [hunk 2]
```diff
@@ -357,6 +371,45 @@ <h3>Locations ([[value.length]])</h3>
...
+                self._setPending(loc_type, false);
+                self._setLocations(loc_type, res.data.results);
+              })
+              .catch(function () {
+                self._setPending(loc_type, false);
+              });
+            return;
+          }
+
+          this.debounce('fetch-locations-' + loc_type, function() {
```
Comment (n1207n) [inline polymer/src/elements/cluster-reporting/indicator-locations-widget.html]: This debounce will always happen whether search text is empty or not. We should put else statement.

Comment (AttilaTheHen) [reply polymer/src/elements/cluster-reporting/indicator-locations-widget.html]: Ahhh, totally missed that - will fix!

File: polymer/src/elements/cluster-reporting/indicator-locations-widget.html (hunk 1 above)
Comment (AttilaTheHen) [reply polymer/src/elements/cluster-reporting/indicator-locations-widget.html]: Will do 👍

File: polymer/src/elements/cluster-reporting/indicator-locations-widget.html [hunk 3]
```diff
@@ -376,38 +376,40 @@ <h3>Locations ([[value.length]])</h3>
...
-              .then(function (res) {
-                self.set('url', res.xhr.responseURL);
-                self._setPending(loc_type, false);
-                self._setLocations(loc_type, res.data.results);
-              })
-              .catch(function () {
-                self._setPending(loc_type, false);
-              });
-            return;
+            this.debounce('fetch-locations-' + loc_type, function() {
```
Comment (n1207n) [inline polymer/src/elements/cluster-reporting/indicator-locations-widget.html]: We could combine 2 this.debounce calls into a single call by moving if-else block into debounce function block.

Comment (AttilaTheHen) [reply polymer/src/elements/cluster-reporting/indicator-locations-widget.html]: 👍 👍

**Pull request merged**